### PR TITLE
Add intents with rule page

### PR DESCRIPTION
**Pull request by joonasroosalung**
New view for adding/editing intents, tied to stories

### CODE REVIEW

File: GUI/src/pages/Training/Intents/index.tsx [hunk 1]
```diff
@@ -457,6 +508,215 @@ const Intents: FC = () => {
...
+    },
+    onError: (error: AxiosError) => {
+      toast.open({
+        type: 'error',
+        title: t('global.notificationError'),
+        message: error.message,
+      });
+    },
+    onSettled: () => {
+      // queryRefresh(selectedIntent?.id || '');
```
Comment (yb-ng) [inline GUI/src/pages/Training/Intents/index.tsx:536]: commented code, consider removing

File: GUI/src/pages/Training/Intents/CommonIntents.tsx [hunk 2]
```diff
@@ -77,20 +76,20 @@ const CommonIntents: FC = () => {
...
+
+    const queryIntent = commonIntents.find(
+      (intent) => intent.id === intentParam
+    );
+
+    if (queryIntent) {
+      setSelectedIntent(queryIntent);
     }
-  }, [commonIntents, searchParams]);
+  }, [intentParam]);
```
Comment (SanderSepp) [inline GUI/src/pages/Training/Intents/CommonIntents.tsx:92]: Can intentsFullList and commonIntents change? If so useEffect is not executed, is it intentional?

**Review by SanderSepp — CHANGES_REQUESTED** [review]

**Review by yb-ng — APPROVED** [review]

File: GUI/src/pages/Training/Intents/CommonIntents.tsx (hunk 2 above)
Comment (SanderSepp) [reply GUI/src/pages/Training/Intents/CommonIntents.tsx:92]: Discussed, all intentional.

**Review by SanderSepp — APPROVED** [review]

**Review by SanderSepp — APPROVED** [review]

**Comment by turnerrainer** [comment]
> ## [Quality Gate Failed](https://sonarcloud.io/dashboard?id=buerokratt_Training-Module&pullRequest=626) **Quality Gate failed**
> Failed conditions  [18.6% Duplication on New Code](https://sonarcloud.io/component_measures?id=buerokratt_Training-Module&pullRequest=626&metric=new_duplicated_lines_density&view=list) (required ≤ 3%)
> 
> [See analysis details on SonarCloud](https://sonarcloud.io/dashboard?id=buerokratt_Training-Module&pullRequest=626)

Will come back to it in later stage.

**Pull request merged**